docs: remove needless scripts

matklad opened the pull request:
* there's no need for `install.sh` when it's just `./scripts/install_zig.sh && ./zig/zig build -Drelease -Dconfig=production`
* there's no need to call `zig build install`, install is the default step
* git hook uses the wrong zig, and in general hooks are very developer-specific

matklad committed:
docs: remove needless scripts

* there's no need for `install.sh` when it's just
  `./scripts/install_zig.sh && ./zig/zig build -Drelease -Dconfig=production`
* there's no need to call `zig build install`, install is the default
  step
* git hook uses the wrong zig, and in general hooks are very
  developer-specific

cb22 commented on README.md:
```diff
@@ -179,7 +179,8 @@ First grab the sources and run the setup script:
 git clone https://github.com/tigerbeetle/tigerbeetle.git
 cd tigerbeetle
-scripts/install.sh
+./scripts/install_zig.sh
```
Ah! Let's point the user to the quick start instructions above? They no longer need source to benchmark.

matklad replied:
I removed this section altogether. I don't think it's possible to meaningfully use these benchmarks unless you know quite a bit more than we are saying here.

cb22 commented on .github/ci/benchmark/tigerbeetle.hcl:
```diff
@@ -99,7 +99,7 @@ git clone ${var.git_url}
 git checkout ${var.git_ref}
 ./scripts/install_zig.sh
-./zig/zig build install -Drelease -Dconfig=production
+./zig/zig build -Drelease -Dconfig=production
```
We can remove this whole folder too, unless you think it's useful to point people to... It's not actively used.

matklad replied:
:+1:

cb22 commented:
(the macOS yaml still references install.sh)

matklad committed:
docs: drop benchmark section from readme

I think it is quiet useless, because it doesn't explain what exactly is
benchmarked and how to interpret the results.

matklad committed:
drop unused benchmarking infrastructure
